src/main/java/net/minecraftforge/common/util/ChunkedTileEntityList.java

```diff
@@ -0,0 +1,283 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2017.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+package net.minecraftforge.common.util;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.Set;
+
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.math.ChunkPos;
+import net.minecraft.world.World;
+
+public class ChunkedTileEntityList implements Collection<TileEntity>
```

Comment:
Already thought about that one, but it wouldn't make a difference in my most cases. It would only change the exception which will be thrown once somebody uses those methods (like get(int), remove(int)).

The only case where it would make a difference would be if a mod casts it to a list (maybe to store it somewhere else). That is rather unlikely, but not impossible. So I'm not really sure about it.

I will leave it for now as it is, to keep it better organized. Maybe someone else knows if this is necessary or not.